### PR TITLE
Fixing and adding more Norwegian oils

### PR DESCRIPTION
**Pull request by knutfrode**
All Norwegian oils are now `gnome_suitable`, after changing `fraction_recovered` `max_value` to `value`.
Added missing cuts to two oils, and added four new oils.

### CODE REVIEW

File: data/oil/NO/NO00159.json [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+                    }
+                ]
+            },
+            "bulk_composition": [
+                {
+                    "value": 2.12,
+                    "unit": "%",
+                    "unit_type": "massfraction"
+                }
+            ]
```
Comment (ChrisBarker-NOAA) [inline data/oil/NO/NO00159.json]: What is this? bulk composition needs a name and a measurment, such as:

```
                {
                    "name": "wax_content",
                    "measurement": {
                        "value": 0.0788,
                        "unit": "fraction",
                        "unit_type": "massfraction"
                    }
                },
```

File: data/oil/NO/NO00160.json [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+                    }
+                ]
+            },
+            "bulk_composition": [
+                {
+                    "value": 1.81,
+                    "unit": "%",
+                    "unit_type": "massfraction"
+                }
+            ]
```
Comment (ChrisBarker-NOAA) [inline data/oil/NO/NO00160.json]: same as 159 -- what is this?

File: data/oil/NO/NO00162.json [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+                        }
+                    }
+                ]
+            },
+            "bulk_composition": [
+                {
+                    "value": 3.38,
+                    "unit": "%",
+                    "unit_type": "massfraction"
+                }
```
Comment (ChrisBarker-NOAA) [inline data/oil/NO/NO00162.json]: and here.

File: data/oil/NO/NO00163.json [hunk 4]
```diff
@@ -0,0 +1,428 @@
...
+                        },
+                        "vapor_temp": {
+                            "value": 705.0,
+                            "unit": "C",
+                            "unit_type": "temperature"
+                        }
+                    }
+                ]
+            },
+            "bulk_composition": [
```
Comment (ChrisBarker-NOAA) [inline data/oil/NO/NO00163.json:272]: and here.

File: data/oil/NO/NO00164.json [hunk 5]
```diff
@@ -0,0 +1,499 @@
...
+                        },
+                        "vapor_temp": {
+                            "value": 688.0,
+                            "unit": "C",
+                            "unit_type": "temperature"
+                        }
+                    }
+                ]
+            },
+            "bulk_composition": [
```
Comment (ChrisBarker-NOAA) [inline data/oil/NO/NO00164.json:344]: and here...

File: data/oil/NO/NO00159.json [hunk 6]
```diff
@@ -0,0 +1,408 @@
...
+    "oil_id": "NO00159",
+    "adios_data_model_version": "0.12.0",
+    "metadata": {
+        "name": "SVALE",
+        "reference": {
+            "year": 2010,
+            "reference": "Sorheim et al. (2010), Kartlegging av forvitringsegenskaper av Norne Blend, Staer, Svale og Alve samt effekt av voksinhibitor paa Norne Blend"
+        },
+	"location": "Norway",
+        "product_type": "crude oil nos",
```
Comment (ChrisBarker-NOAA) [inline data/oil/NO/NO00159.json]: This should be "Crude Oil NOS" -- case sensitive.

**Review by ChrisBarker-NOAA — CHANGES_REQUESTED** [review]
A few issues -- highlighted above, and when I run the validation, I get:

`/Users/chris.barker/Hazmat/GitLab/noaa-oil-data/data/oil/NO`

**Generated:** Jun 28, 2024 - 14:00


#### `NO00022`: ELLI SOUTH

E062: Viscosity data is not strictly decreasing


#### `NO00104`: VISUND

E062: Viscosity data is not strictly decreasing

W100: Not GNOME compatible: Viscosities must be decreasing with temperature


#### `NO00109`: SVALIN

E060: Oil fraction in distillation cuts is not accumulative

I haven't looked carefully at what the issues are myself.

## NOTE:

you can run the validation yourself, with the adiso_db library -- it isntalls a command line utility:

`adios_db_validate [dir_with_data]`

and it will generate two reports:

`validation_by_record.md`

`validation_by_error.md`

same information, just in different order

File: data/oil/NO/NO00159.json (hunk 6 above)
Comment (knutfrode) [reply data/oil/NO/NO00159.json]: This is fixed in [8f4c482](https://github.com/NOAA-ORR-ERD/noaa-oil-data/pull/3/commits/8f4c4822d95693e1573065005e8ed1650d2a1c3e)

File: data/oil/NO/NO00164.json (hunk 5 above)
Comment (knutfrode) [reply data/oil/NO/NO00164.json:344]: These are also fixed in [8f4c482](https://github.com/NOAA-ORR-ERD/noaa-oil-data/pull/3/commits/8f4c4822d95693e1573065005e8ed1650d2a1c3e)

**Comment by knutfrode** [comment]
@ChrisBarker-NOAA 
I believe I have now fixed all these errors, so that these oils should all be gnome compatible.

**Comment by knutfrode** [comment]
For me this PR is now ready to merge, and I believe the requested changes are fulfilled.

**Comment by ChrisBarker-NOAA** [comment]
Thanks! I'll try to get this merged in soon -- it's a bit tricky, as the "official" version is on an internal git instance, but I'll work it out.

**Comment by ChrisBarker-NOAA** [comment]
I had a chance to take a look -- there are still some issues -- we added some more validation recently.

[validation_by_error.md](https://github.com/user-attachments/files/17758645/validation_by_error.md)

```
#### `NO00150 (NO00150)` -- SF NORD BRENT:

Duplicate sub_sample name(s): ['16.0% Evaporated', '28.8% Evaporated', '38.7% Evaporated'] names must be unique
```
This one looks to me, at a glance, like the data somehow got split up between two subsamples.

```
## E062: (5 records affected)

#### `NO00022 (NO00022)` -- ELLI SOUTH:

Viscosity data is not strictly decreasing
```
This one is checking if the viscosity goes down as the temperature goes up -- which it should.

If it doesn't, it means one of:
- an error / typo
- viscosity was measured by two different labs / different methods
- viscosity was measured at different shear rates
   - if that's the case, the shear rate can be specified, if you know what it is, and teh code will work it out.

**Pull request closed without merging**